test/unit/test_pandas_extensions.py

```diff
@@ -268,3 +277,49 @@ def test_pandas_arrays_describe():
         {"array": [3, 3, 0, 1]}, index=["count", "unique", "top", "freq"]
     )
     assert_frame_equal(describe, data.describe())
+
+
+def test_infer_custom_types_returns_empty_data_frame_for_empty_data_frame():
+    inferred = _infer_custom_types(pd.DataFrame())
+    assert_frame_equal(pd.DataFrame(), inferred)
+
+
+def test_infer_custom_types_returns_same_data_frame():
+    data = pd.DataFrame({"col1": [1, 2, 3], "col2": ["x", "y", "z"]})
+    inferred = _infer_custom_types(data)
+    assert_frame_equal(data, inferred)
+
+
+def test_infer_custom_types_infers_images_type(tmp_path):
+    data = pd.DataFrame({"col1": [1, 2, 3], "img": [_image(1), _image(2), _image(3)]})
+    inferred = _infer_custom_types(data)
+    parquet_path = tmp_path / str(uuid4())
+    data.to_parquet(parquet_path, engine=_PARQUET_ENGINE)
+    data_parquet = pd.read_parquet(parquet_path, engine=_PARQUET_ENGINE)
```

Comment:
why does *this* unit test have this logic? shouldn't we only test `infer_custom_types`? (same for other tests below)

especially given we already have `test_pandas_images_read_write` and `test_pandas_arrays_read_write`.

Reply:
To make sure data frame is still written and read correctly after the implicit type conversion. I put a comment for duplication